third-party/qthread/qthread-src/src/alloc/chapel.c

```diff
@@ -42,10 +46,36 @@ void qt_internal_alignment_init(void) {
 
 void *qt_internal_aligned_alloc(size_t        alloc_size,
                                      uint_fast16_t alignment) {
-    return chpl_memalign(alignment, alloc_size);
+  void *ret;
+
+  assert(alloc_size > 0);
+  switch (alignment) {
+    case 0:
+      ret = chpl_mem_alloc(alloc_size, CHPL_RT_MD_TASK_LAYER_UNSPEC,
+                           0, CHPL_FILE_IDX_INTERNAL);
+      break;
+#if defined(HAVE_16ALIGNED_MALLOC)
```

Comment:
The `configure` check explicitly checks the alignment of `malloc()` and not the wrappers , so I think this is correct. I also don't think this special casing is worthwhile, I'ld just have this function be:

```c
return chpl_mem_memalign(alignment, alloc_size,
                         CHPL_RT_MD_TASK_LAYER_UNSPEC,
                         0, CHPL_FILE_IDX_INTERNAL);
```

Reply:
And jemalloc is really optimized. I can't imagine the branch for the size here will be any faster than just calling the jemalloc aligned interface (and qthreads is pooling anyways, so this routine won't get called much). I'd be in favor of just reverting your second commit that added this fast-path

Reply:
I could paraphrase from the 3rd commit's message into the PR message -- I think it may be worthwhile to keep this because one could imagine that `malloc()` for small allocs might be faster than `memalign()`.

Reply:
(and also that that code is duplicated from the regular Qthreads mem shim `src/alloc/base.c`)

Reply:
Hmm, good points (jemalloc highly optimized, this not called much).  Okay, I'll yank it back out.

Reply:
While I don't think there's much of a perf win here, I was more concerned about the correctness issue (HAVE_16ALIGNED_MALLOC only being valid for `malloc()` and not for our memory allocator).